src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XSSInImgTagAttribute.java

```diff
@@ -168,18 +176,18 @@ public UrlParamBasedImgTagAttrInjection() {
             variant = Variant.SECURE,
             htmlTemplate = "LEVEL_1/XSS")
     public ResponseEntity<String> getVulnerablePayloadLevelSecure(
-            @RequestParam Map<String, String> queryParams) {
+            @RequestParam(PARAMETER_NAME) String imageLocation) {
         String vulnerablePayloadWithPlaceHolder = "<img src=\"%s\" width=\"400\" height=\"300\"/>";
-        StringBuilder payload = new StringBuilder();
 
-        for (Map.Entry<String, String> map : queryParams.entrySet()) {
-            if (allowedValues.contains(map.getValue())) {
-                payload.append(
-                        String.format(
-                                vulnerablePayloadWithPlaceHolder,
-                                StringEscapeUtils.escapeHtml4(map.getValue())));
-            }
+        if (imageLocation.startsWith(IMAGE_RESOURCE_PATH)
```

Comment:
can you please add another level with validation against `allowedValues` along with this one?

Reply:
can you please address this one and then we can close this PR.